### PR TITLE
Support backup restores on Chromebook

### PR DESCRIPTION
**Pull request by mikehardy**
## Pull Request template

Please, go through these checks before you submit a PR.

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

## Purpose / Description
Chromebook has an emulator now which makes testing much easier.
This PR has two commits - one detects the emulator as a Chromebook, the second one allows for backups to be restored since that works on Chromebook (I tested it)

## How Has This Been Tested?

https://developer.android.com/topic/arc/emulator

Ran the app on the emulator with an adb install and noticed backup restores worked.
Logged into the emulator with adb and ran getprops to find something I could identify it with, rebuilt and tested.

## Learning (optional, can help others)

I'm working on figuring out how to push new releases easily with a question to the previous contributor on #3906 otherwise I'll figure it out in a while

### CODE REVIEW

**Comment by mikehardy** [comment]
They've always had storage apparently, it's just managed differently. Basically you can't rely on it being in a particular spot and you can't rely on it not being erased at any time, but for a picker, and for showing that there may be backups to select, it's sufficient. I don't have a link to show it's possible other than before I had the emulator detected correctly my test build wasn't guarded by chromebook checks, so I got the full deckpicker tools menu, and backups worked fine - every time you restore it actually takes a backup first so every time I checked it I had another entry in the list...

**Comment by mikehardy** [comment]
oh - and the reason import won't work that way is because the apps are containerized, so one app's storage won't work for another, so you can't download from chrome to the place where ankidroid would see files. A picker would allow for google docs access though, and maybe other locations they expose in the default picker, so sharing is possible but only if it wasn't restricted to what the app thinks is '/sdcard/AnkiDroid'

**Comment by mikehardy** [comment]
You asked for a link on ChromeOS allowing filesystem access, I did a quick deep dive. My previous statement that cross-app storage wasn't possible was incorrect.

Storage on ChromeOS:
- everything runs inside the "ARC" (Android Runtime Container) which used to be more dynamic with the "ARC Welder" Chrome App that is still in use and where "Chrome Apps" run, but is now being built into the system and made available by default in the new "Play Store on ChromeOS" style
- ARC uses linux kernel namespaces to containerize Android Apps
- As ChromeOS does ARC container creation / ARC setup, the ChromeOS user's cryptohome is mounted in various places inside the ARC to provide filesystem layout that Android app expects, one of those is an 'sdcard' area that is readable and writable to Android apps, and shared with the ChromeOS file space 

So any Android app the user has installed will share the ARC environment, and thus /data and /sdcard storage space, and CardContentProvider etc on new ChromeOS. Additionally, the system file picker in ChromeOS and ARC can see the same Downloads folder, so they do have cross-container file visibility as well (I personally tested this).

Files in ChromeOS (and thus bind-mounted into ARC) aren't guaranteed to be retained as the ChromeOS manages space itself, but permission to use the space certainly exists and for personal chromebooks users can control retention with behavior.

In short:
- backups/restores function with only caveat being backups may be trimmed by system
- a filepicker would let .apkg import work, from a space shared with the main OS, current method of getting a file into /sdcard/AnkiDroid isn't possible by default from ChromeOS, though it would likely be possible from an ARC app 
- A port of the CSV import would definitely enable Chromebook/Android-only people to function (https://github.com/ankidroid/Anki-Android/issues/3517#issuecomment-167377628 (people can use Sheets, export as .csv to Downloads, then open in Anki from Downloads)

Full details, and I have to admit this is a cool solution for Android app isolation in a multi-user environment:
https://chromium.googlesource.com/chromiumos/platform2/+/master/arc/container-bundle/

**Comment by mikehardy** [comment]
@timrae I pushed a couple new commits, because some chromebooks have cameras, so it's a hardware check with manifest declaration, not a device-type check. And we weren't doing camera access safely which may be causing crashes, so I protected it [per API recommendations](https://developer.android.com/training/camera/photobasics#TaskManifest) and with permissions checking.

**Comment by timrae** [comment]
So this wants to get merged right?

**Comment by mikehardy** [comment]
Yes, it fixes a crash I actually experienced and makes camera use more standards conformant

**Comment by Jd-75** [comment]
Just want download

**Comment by david-allison-1** [comment]
> Just want download

https://play.google.com/store/apps/details?id=com.ichi2.anki - should now work for Chromebooks
